### PR TITLE
Event items controller

### PR DESCRIPTION
**Pull request by SemraChernet**
Added Event Category API.

### CODE REVIEW

File: EventCatalogAPI/Controllers/EventItemsController.cs [hunk 1]
```diff
@@ -16,6 +17,10 @@ public class EventItemsController : ControllerBase
     {
         private readonly EventContext _context;
         private readonly IConfiguration _config;
+        private object eventCategoryId;
+        private object eventCategoryID;
```
Comment (misurac) [inline EventCatalogAPI/Controllers/EventItemsController.cs:21]: I don't think we need lines 20 and 21

File: EventCatalogAPI/Controllers/EventItemsController.cs [hunk 2]
```diff
@@ -63,5 +68,34 @@ public async Task<IActionResult> Dates()
...
+            return Ok(events);
+        }
+
+
+        //Sort Event by Category 
+        [HttpGet]
+        [Route("[action]")]
+        public async Task<IActionResult> EventCategories(
+            [FromQuery] int pageIndex = 0,
+            [FromQuery] int pageSize = 4)
```
Comment (misurac) [inline EventCatalogAPI/Controllers/EventItemsController.cs:87]: We will need to accept an input category FromQuery - something like this: "[FromQuery] string eventCategory".

File: EventCatalogAPI/Controllers/EventItemsController.cs [hunk 3]
```diff
@@ -63,5 +68,34 @@ public async Task<IActionResult> Dates()
...
+        {
+
+            var events = await _context.EventItems
+                    .OrderBy(c => c.EventCategory)
```
Comment (misurac) [inline EventCatalogAPI/Controllers/EventItemsController.cs:91]: We need to add a Where statement before the OrderBy using the eventCategory we got from the query - something like this: ".Where(c => c.EventCategory == category)"

**Comment by LalithaNadimpalli** [comment]
You can replace eventTyeID with CategoryID from my code in filters:)

**Pull request merged**